packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts

```diff
@@ -401,65 +495,51 @@ export const useSpinButtonStyles_unstable = (state: SpinButtonState): SpinButton
   const buttonStyles = useButtonStyles();
   const buttonDisabledStyles = useButtonDisabledStyles();
+  const inputClassName = useInputClassName();
   const inputStyles = useInputStyles();
 
-  // Grab the root className here so we can be sure to merge is last
-  const rootClassName = state.root.className;
-  state.root.className = undefined;
-  // Reuse react-input's styles without re-using the Input component.
-  useInputStyles_unstable({
-    size,
-    appearance,
-    input: state.input,
-    root: state.root,
-    components: {
-      root: 'span',
-      input: 'input',
-      contentBefore: 'span',
-      contentAfter: 'span',
-    },
-  });
-
   state.root.className = mergeClasses(
-    state.root.className, // Get the classes from useInputStyles_unstable
     spinButtonClassNames.root,
-    rootStyles.base,
+    useRootClassName(),
     rootStyles[size],
-    appearance === 'outline' && rootStyles.outline,
-    appearance === 'underline' && rootStyles.underline,
+    rootStyles[appearance],
     filled && rootStyles.filled,
     !disabled && appearance === 'outline' && rootStyles.outlineInteractive,
     !disabled && appearance === 'underline' && rootStyles.underlineInteractive,
     !disabled && filled && rootStyles.filledInteractive,
     !disabled && invalid && rootStyles.invalid,
     disabled && rootStyles.disabled,
-    rootClassName, // Make sure any original class name is applied last
+    state.root.className,
   );
 
   state.incrementButton.className = mergeClasses(
     spinButtonClassNames.incrementButton,
     spinState === 'up' && `${spinButtonExtraClassNames.buttonActive}`,
-    buttonStyles.base,
-    buttonStyles.incrementButton,
+    useIncrementButtonClassName(),
     buttonStyles[appearance],
-    size === 'small' ? buttonStyles.incrementButtonSmall : buttonStyles.incrementButtonMedium,
+    size === 'small' && buttonStyles.incrementButtonSmall,
     (atBound === 'max' || atBound === 'both') && buttonDisabledStyles.base,
     (atBound === 'max' || atBound === 'both') && buttonDisabledStyles[appearance],
     state.incrementButton.className,
   );
   state.decrementButton.className = mergeClasses(
     spinButtonClassNames.decrementButton,
     spinState === 'down' && `${spinButtonExtraClassNames.buttonActive}`,
-    buttonStyles.base,
-    buttonStyles.decrementButton,
+    useDecrementButtonClassName(),
     buttonStyles[appearance],
-    size === 'small' ? buttonStyles.decrementButtonSmall : buttonStyles.decrementButtonMedium,
+    size === 'small' && buttonStyles.decrementButtonSmall,
     (atBound === 'min' || atBound === 'both') && buttonDisabledStyles.base,
     (atBound === 'min' || atBound === 'both') && buttonDisabledStyles[appearance],
     state.decrementButton.className,
   );
 
-  state.input.className = mergeClasses(spinButtonClassNames.input, state.input.className, inputStyles.base);
+  state.input.className = mergeClasses(
+    spinButtonClassNames.input,
+    inputClassName,
```

Comment:
(nit) for consistency
```suggestion
    useInputClassName(),
```